File: app/src/main/java/com/pkfcooperparry/template/ui/form/page/tab/TabFragment.java
package com.pkfcooperparry.template.ui.form.page.tab;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.LinearLayout;

import com.pkfcooperparry.template.BR;
import com.pkfcooperparry.template.R;
import com.pkfcooperparry.template.databinding.FragmentTabBinding;
import com.pkfcooperparry.template.ui.base.BaseFragment;
import com.pkfcooperparry.template.utils.Navigator;

import javax.annotation.Nullable;
import javax.inject.Inject;

public class TabFragment extends BaseFragment<FragmentTabBinding, TabFragmentViewModel> implements TabFragmentViewInterface {
    @Inject
    Navigator navigator;
    @Inject
    TabFragmentViewModel tabFragmentViewModel;


    FragmentTabBinding fragmentSpecificStatsBinding;
    LinearLayout linearLayout;

    public static TabFragment newInstance(long tabId) {
        Bundle args = new Bundle();
        TabFragment fragment = new TabFragment();
        args.putLong("tabId", tabId);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public int getBindingVariable() {
        return BR.viewModel;
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_tab;
    }

    @Override
    public TabFragmentViewModel getViewModel() {
        return tabFragmentViewModel;
    }


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        tabFragmentViewModel.setViewInterface(getActivity(), this);

    }

    @Override
    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        fragmentSpecificStatsBinding = getViewDataBinding();
        linearLayout = view.findViewById(R.id.parent);
        setupUi();

    }

    @Override
    public void setupUi() {

    }


}

File: app/src/main/java/com/pkfcooperparry/template/ui/form/page/tab/TabFragment.kt
package com.pkfcooperparry.template.ui.form.page.tab

import android.os.Bundle
import android.view.View
import android.widget.LinearLayout

import com.pkfcooperparry.template.BR
import com.pkfcooperparry.template.R
import com.pkfcooperparry.template.databinding.FragmentTabBinding
import com.pkfcooperparry.template.ui.base.BaseFragment
import com.pkfcooperparry.template.utils.Navigator
import javax.inject.Inject

class TabFragment : BaseFragment<FragmentTabBinding, TabFragmentViewModel>(), TabFragmentViewInterface {
    var navigator: Navigator? = null
        @Inject set

    var tabFragmentViewModel: TabFragmentViewModel? = null
        @Inject set

    internal var fragmentSpecificStatsBinding: FragmentTabBinding? = null
    internal lateinit var linearLayout: LinearLayout

    override val bindingVariable: Int
        get() = BR.viewModel

    override val layoutId: Int
        get() = R.layout.fragment_tab

    override fun getViewModel(): TabFragmentViewModel {
        return this!!.tabFragmentViewModel!!
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activity?.let { tabFragmentViewModel!!.setViewInterface(it, this) }

    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        fragmentSpecificStatsBinding = viewDataBinding
        linearLayout = view.findViewById(R.id.parent)
        setupUi()

    }

    override fun setupUi() {

    }

    companion object {

        fun newInstance(tabId: Long): TabFragment {
            val args = Bundle()
            val fragment = TabFragment()
            args.putLong("tabId", tabId)
            fragment.arguments = args
            return fragment
        }
    }


}
